### PR TITLE
Search results for "UTF-8 with BOM" files shifted on first line by a character

### LINKED ISSUE #66188: Search results for "UTF-8 with BOM" files shifted on first line by a character again
Issue Type: <b>Bug</b>

1. Open a folder containing UTF-8 with BOM files
2. Search something that on the first line in Search View
image

VS Code version: Code - Insiders 1.31.0-insider (28e7467661d67c05c3662ff900ee1984f7e2fbcc, 2019-01-07T06:20:02.775Z)
OS version: Windows_NT x64 10.0.17134

### PR DESCRIPTION
**Pull request by ghost**
Fix #66188 

This is my first pull request in this project. Please let me know if my solution is not good enough, I am willing to improve it.

### CODE REVIEW

File: src/vs/workbench/parts/search/common/searchModel.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class Match {
 	private _fullPreviewRange: ISearchRange;
 
 	constructor(private _parent: FileMatch, private _fullPreviewLines: string[], _fullPreviewRange: ISearchRange, _documentRange: ISearchRange) {
-		this._oneLinePreviewText = _fullPreviewLines[_fullPreviewRange.startLineNumber];
+		this._oneLinePreviewText = stripUTF8BOM(_fullPreviewLines[_fullPreviewRange.startLineNumber]);
```
Comment (roblourens) [inline src/vs/workbench/parts/search/common/searchModel.ts]: Why this?

File: src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts [hunk 2]
```diff
@@ -265,10 +265,10 @@ export class RipgrepParser extends EventEmitter {
 				stats.lastLineLength :
 				stats.lastLineLength + startCol;
 
-			if (lineNumber === 0 && i === 0 && startsWithUTF8BOM(matchText)) {
+			if (lineNumber === 0 && i === 0 && startsWithUTF8BOM(fullText)) {
 				matchText = stripUTF8BOM(matchText);
-				startCol -= 3;
-				endCol -= 3;
+				startCol -= 1;
```
Comment (roblourens) [inline src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: Why change this? The BOM is 3 bytes long.

File: src/vs/workbench/parts/search/common/searchModel.ts (hunk 1 above)
Comment (roblourens) [reply src/vs/workbench/parts/search/common/searchModel.ts]: But the BOM should be stripped already by ripgrepTextSearchEngine right?

Comment (victoriqueko) [reply src/vs/workbench/parts/search/common/searchModel.ts]: I can move it to ripgrepTextSearchEngine.

**Comment by victoriqueko** [comment]
I found another solution: Remove `&& options.encoding !== 'utf8'`

https://github.com/Microsoft/vscode/blob/f0f3b922bcb081c6488b7299dbef4076a1cfde82/src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts#L333

then ripgrep will not output BOM character in JSON.

Or I can manually strip the BOM character in ripgrepTextSearchEngine.

Which do you prefer?

**Comment by roblourens** [comment]
I want to keep that because I found that search was faster in some cases without it. Also, you could probably search with a different encoding but still end up finding results in UTF8 files with BOMs. Also I don't trust ripgrep to keep that behavior forever. So, let's keep the check to strip the BOM on our end.

File: src/vs/workbench/parts/search/common/searchModel.ts (hunk 1 above)
Comment (roblourens) [reply src/vs/workbench/parts/search/common/searchModel.ts]: But it's already there, right? That file should be stripping the BOM correctly in all cases with your change.

**Comment by victoriqueko** [comment]
I strip `fullText` at https://github.com/Microsoft/vscode/blob/c22caf616a9d6baa90ab904292bddc14a7b09ffd/src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts#L285 now. So it's not need to edit searchModel.ts.

**Comment by roblourens** [comment]
Looks great, thanks!

**Pull request merged**